### ISSUE TITLE
Changing number of neurons in the hidden layer

### ISSUE THREAD

**Issue by DaMexicanJustice** [HelgeCPH/get_things_done_with_python#5]
I am at my wit's end after trying to change the number of neurons in the hidden layer. From my understanding, where we need to have a look (what we change to change the # of neurons) is the following lines:
`# correct answers will go here
Y_ = tf.placeholder(tf.float32, [None, 10])
# weights W[784, 10]   784=28*28
W = tf.Variable(tf.zeros([784, 10]))
# biases b[10]
b = tf.Variable(tf.zeros([10]))`

We can spot a value '10' used and supposedly we need to change these to another number to change the number of neurons that we are working with. However, changing the 10s to 20s for instance, will cause a value error exception complaining about a mismatch in dimensions. The culprits involved are ones above as well as:
`Y = tf.nn.softmax(tf.matmul(XX, W) + b)
cross_entropy = -tf.reduce_mean(Y_ * tf.log(Y)) * 1000.0
for i in range(2000+1): training_step(i, i % 50 == 0, i % 10 == 0)
a, c, im, w, b = sess.run([accuracy, cross_entropy, I, allweights, allbiases], feed_dict={X: batch_X, Y_: batch_Y})`

I have traced the error to occur in those lines of code. It complains about adding unequal dimensions and then about multiplication once fixing the add one by changing the 10s into 20s where I can. I'm guessing the problems lie with matrix/vector multiplication somewhere? I don't understand at all how to change # of neurons though :(

**Comment by HelgeCPH** [comment]
Hej,

You likely do not want to change the number `10` to something different than `10` for a neural network classifying the digits of the MNIST dataset. Try to understand -with the help of your group mates- why. Hint, how many digits do we want to classify?

See step number 7 in the tutorial (https://codelabs.developers.google.com/codelabs/cloud-tensorflow-mnist/#6) to see how to add another layer of perceptrons to your network.

**Comment by DaMexicanJustice** [comment]
In your answer to my question of what to do for this assignment, you asked us to change the number of perceptrons in the hidden layer AND add a second layer. Is this two different things or one and the same?

**Comment by HelgeCPH** [comment]
What do your group members think about this question? And what is your opinion about it?
